### DISCUSSION TITLE
How to set isValid to false after onSubmit

### DISCUSSION THREAD

**Discussion by kennethslee613** [react-hook-form/react-hook-form#4998] [General]
I am currently using the onSubmit mode for both mode and reValidateMode. When I submit the form, it runs the validations, but if one of the validations fail, isValid is still set to true even though the errors are working properly. I see in the documentation that isValid only works for onChange and onBlur, so I was wondering how I can set isValid to false after submitting the form and running into a failed validation?

**Comment by bluebill1049** [comment]
we don't support `isValid` for onSubmit mode, isVaid required to check on each onChange or onBlur

**Reply by kennethslee613** [reply]
is there a way to check if the form is valid after i submit using the library? or do i just have to create my own variable to keep track of it

**Reply by bluebill1049** [reply]
IsSumbitSucessful state
